lua/entities/gmod_wire_expression2/core/custom/sv_rune2.lua

```diff
@@ -20,69 +21,54 @@ local function buildBody(args) -- Why does the wireteam do this
 end
 
 -- We use this for try and catch
-local function runE2InstanceSafe(compiler,func,body) -- Varargs to pass to the e2 function
-    -- Will always return pcallerror,errstr first even if it didn't error.
-    local args = {pcall(func,compiler,body)}
+local function runE2InstanceSafe(compiler,func,body,...)
+    -- Always return pcallerror,errstr first even if it didn't error.
+    local args = {pcall(func,compiler,body,...)}
     local success = table_remove(args,1)
     if success then
-        return true,"none",args
-    else
-        -- Don't return args, that would be a waste
-        local errmsg = table_remove(args,1)
-        if errmsg == "exit" then return true,"none",{} end -- nice exit()
-        if errmsg == "perf" then errmsg = "tick quota exceeded" end
-        return false,errmsg
+        return true,nil,args
     end
+    -- If unsuccessful, don't return args, that would be a waste
+    local errmsg = table_remove(args,1)
+    if errmsg == "exit" then return true,nil,{} end -- nice exit()
+    if errmsg == "perf" then errmsg = "tick quota exceeded" end
+    return false,errmsg
 end
 
-local function getE2FuncFromStr(compiler,funcname)
-    local funcs = compiler.funcs
-    local e2func = funcs[funcname]
-    if not e2func then -- We will look for any udfs that have the name before the parenthesis.
-        local fncnameproper = funcname:match("(%w*)%(") or funcname
-        for K,V in pairs(funcs) do
-            local proper = K:match("(%w*)%(")
-            if proper and string.find(proper,fncnameproper) then return V end
-        end
-    else
-        return e2func
-    end
-end
+-- TODO: Set E2 costs ( __e2setcost(N) ).
 
 
 -- Literally like pcall()
--- Returns array, first argument is a number stating whether the function executed successfully, rest are varargs.
-
-e2function array try(string try) -- If you *really* want to pass arguments to the function then just make another function that calls that function.
-    local tryfnc = getE2FuncFromStr(self,try)
-    if not tryfnc then error("Try called without an existing try function ["..try.."]",0) end
-    local success,errstr,args = runE2InstanceSafe(self,tryfnc)
+-- Returns table, first argument is a number stating whether the function executed successfully, rest are varargs.
+-- TODO: variadic/varargs support [see array(...) / table(...) / select(N...) E2 functions for reference]
+e2function table try(string try)
```

Comment:
I have changed the return type from `array` to `table`, because using an `array` was a really bad design decision. Here's why:
[E2 array **can not** hold another array/table/gtable]() and that's the problem, because this means if you were to call your UDF which returns a `table` value, then you won't be able to access that table.
- [x] Fix the `try` function. Change its return type from `array` to `table` to allow arrays/tables to be returned from UDF.

Reply:
Also, such a function should not be throwing an error by design!
- [x] Made `try` not throw an error if the UDF is not defined.

Reply:
Originally I made this function in mind of the defined() func, since you could just check for yourself. Ig this is fine